generator/src/googleapis/codegen/template_objects.py

```diff
@@ -128,6 +128,35 @@ def __init__(self, def_dict, api, parent=None, wire_name=None,
       self.SetTemplateValue('description',
                             self.ValidateAndSanitizeComment(self.StripHTML(d)))
 
+  def ComputeNonDuplicatedName(self, class_name):
+    parent_path = self.parentPath()
+    occurrences = 0
+
+    def increment_name(s, n):
+      """
+      Increments the number (if any) of the duplicated class name
+      :param s: class_name
+      :param n: occurrences
+      :return: new class_name with incremented occurrence (e.g. Test1 to Test2
+      or Test to Test1)
+      """
+      return s[0: - len(str(n - 1)) if n > 1 else len(s)] + str(n), n
+
+    # increase occurrences as they occur in the path
+    # from root class to current one
+    for parent_class_name in parent_path:
+      if class_name == parent_class_name:
+        class_name, occurrences = increment_name(class_name, occurrences + 1)
+
+    # increase occurrences as they occur in sibling resources
+    if self.parent is not None:
+      siblings = self.parent.children
+      for sibling in siblings:
+        if sibling.GetTemplateValue('className') == class_name:
+          class_name, occurrences = increment_name(occurrences, occurrences + 1)
+
+    return class_name
```

Comment:
It's usually not a best practice to change the value of a argument and return it back. It's better to have something like `new_class_name` initialized in the beginning of the method and change it instead.

Reply:
using a copy of the variable at the beginning of the method